test/functional/p2p_node_network_limited.py

```diff
@@ -46,6 +49,42 @@ def setup_network(self):
         self.add_nodes(self.num_nodes, self.extra_args)
         self.start_nodes()
 
+    def test_avoid_requesting_old_blocks(self):
+        self.log.info("Test node not requesting blocks beyond limited peer threshold")
+        # The request of blocks further than the NETWORK_NODE_LIMITED threshold causes a direct disconnection.
+        pruned_node = self.nodes[0]
+        miner = self.nodes[1]
+        full_node = self.nodes[2]
+
+        # Connect and generate block to ensure IBD=false
+        self.connect_nodes(1, 2)
+        self.connect_nodes(0, 2)
```

Comment:
The test would fail if run standalone without the preceding tests. The failure is on line 78: `self.sync_blocks([miner, pruned_node])`. That is because the test relies on the pruned node being connected to the miner which is indeed the case - some leftover from previous tests. This diff makes it succeed when run standalone:

```diff
--- i/test/functional/p2p_node_network_limited.py
+++ w/test/functional/p2p_node_network_limited.py
@@ -54,25 +54,24 @@ class NodeNetworkLimitedTest(BitcoinTestFramework):
         # The request of blocks further than the NETWORK_NODE_LIMITED threshold causes a direct disconnection.
         pruned_node = self.nodes[0]
         miner = self.nodes[1]
         full_node = self.nodes[2]
 
         # Connect and generate block to ensure IBD=false
+        self.connect_nodes(0, 1)
         self.connect_nodes(1, 2)
-        self.connect_nodes(0, 2)
         self.generate(miner, 1, sync_fun=self.no_op)
 
         # Verify all peers are at the same height
         self.sync_blocks([miner, pruned_node])
                                      
         # Verify peers are out of IBD
         for node in self.nodes:
             assert not node.getblockchaininfo()['initialblockdownload']    
 
         # Disconnect full_node from the other two peers (the node will remain out of IBD)
-        self.disconnect_nodes(0, 2)
         self.disconnect_nodes(1, 2)                                       
         assert_equal(len(full_node.getpeerinfo()), 0)                     
 
         # Mine blocks and sync the pruned node. Surpass the NETWORK_NODE_LIMITED threshold.
```

It is also shorter and I think more clear.

Reply:
Ended up doing it in a slightly different way. Using `setnetworkactive` to isolate the full node from the other two. I think that is clearer in this way.